File: 2429-minimize-xor/2429-minimize-xor.py
class Solution:
    def minimizeXor(self, num1: int, num2: int) -> int:
        count = bin(num2).count('1')
        ans = 0
        for i in range(31, -1, -1):
            if count > 0 and (num1 & (1 << i)):
                ans |= (1 << i)
                count -= 1
        for i in range(32):
            if count > 0 and (ans & (1 << i)) == 0:
                ans |= (1 << i)
                count -= 1
        return ans

File: LeetCode/2429-minimize-xor/2429-minimize-xor.cpp
class Solution {
public:
    int minimizeXor(int num1, int num2) {
        int count = 0;
        int ans = 0;
        for (int i = 0; i < 32; i++) {
            if (num2 & 1 << i) {
                count++;
            }
        }
        for (int i = 31; i >= 0 && count > 0; i--) {
            if ((1 << i) & num1) {
                count--;
                ans = ans | 1 << i;
            }
        }
        for (int i = 0; i < 32 && count > 0; i++) {
            if ((1 << i & ans) == 0) {
                ans = ans | 1 << i;
                count--;
            }
        }
        return ans;
    }
};